src/LiveQuery/ParseLiveQueryServer.js

```diff
@@ -374,98 +403,149 @@ class ParseLiveQueryServer {
     return matchesQuery(parseObject, subscription.query);
   }
 
-  _matchesACL(acl: any, client: any, requestId: number): any {
+  getAuthForSessionToken(
+    sessionToken: ?string
+  ): Promise<{ auth: ?Auth, userId: ?string }> {
+    if (!sessionToken) {
+      return Promise.resolve({});
+    }
+    const fromCache = this.authCache.get(sessionToken);
+    if (fromCache) {
+      return fromCache;
+    }
+    const authPromise = getAuthForSessionToken({
+      cacheController: this.cacheController,
+      sessionToken: sessionToken,
+    })
+      .then(auth => {
+        return { auth, userId: auth && auth.user && auth.user.id };
+      })
+      .catch(() => {
+        // If you can't continue, let's just wrap it up and delete it.
+        // Next time, one will try again
+        this.authCache.del(sessionToken);
+        return {};
```

Comment:
@westhom This is why the CPU is spinning like a madman on invalid tokens. here we delete the entry in the cache which forces the session to be retried. If the session token is invalid, it's unlikely it will become valid any time soon.

@acinader @dplewis what do you think? should we 
- continue with a default auth and no user ID and expire after a shorter period like 5 minutes?
- cut the connection
- reject connections where the auth token is invalid?

What would be best?